client/src/components/Form/FormElement.vue

```diff
@@ -1,270 +1,353 @@
-            </div>
-            <span v-else class="ui-form-title-text">{{ title }}</span>
+            </span>
+            <span v-else class="ui-form-title-text">{{ props.title }}</span>
+
+            <span
+                v-if="isRequired && isRequiredType"
+                v-b-tooltip.hover
+                class="ui-form-title-star"
+                title="required"
+                :class="{ warning: isEmpty }">
+                *
+                <span v-if="isEmpty" class="ui-form-title-message warning"> required </span>
+            </span>
+            <span v-else-if="isRequiredType" class="ui-form-title-message"> - optional </span>
         </div>
-        <div v-if="showField" class="ui-form-field" :data-label="title">
-            <FormBoolean v-if="type == 'boolean'" :id="id" v-model="currentValue" />
-            <FormHidden v-else-if="isHiddenType" :id="id" v-model="currentValue" :info="attrs['info']" />
+
+        <div v-if="showField" class="ui-form-field" :data-label="props.title">
+            <FormBoolean v-if="props.type === 'boolean'" :id="props.id" v-model="currentValue" />
+            <FormHidden v-else-if="isHiddenType" :id="props.id" v-model="currentValue" :info="attrs['info']" />
             <FormNumber
-                v-else-if="type == 'integer' || type == 'float'"
-                :id="id"
+                v-else-if="props.type === 'integer' || props.type === 'float'"
+                :id="props.id"
                 v-model="currentValue"
                 :max="attrs.max"
                 :min="attrs.min"
                 :type="type"
                 :workflow-building-mode="workflowBuildingMode" />
-            <FormColor v-else-if="type == 'color'" :id="id" v-model="currentValue" />
-            <FormDirectory v-else-if="type == 'directory_uri'" v-model="currentValue" />
+            <FormColor v-else-if="props.type === 'color'" :id="props.id" v-model="currentValue" />
+            <FormDirectory v-else-if="props.type === 'directory_uri'" v-model="currentValue" />
             <FormParameter
                 v-else-if="backbonejs"
-                :id="id"
-                ref="params"
+                :id="props.id"
                 v-model="currentValue"
-                :data-label="title"
-                :type="type"
+                :data-label="props.title"
+                :type="props.type"
                 :attributes="attrs" />
-            <FormInput v-else :id="id" v-model="currentValue" :area="attrs['area']" />
+            <FormInput v-else :id="props.id" v-model="currentValue" :area="attrs['area']" />
         </div>
-        <div v-if="showPreview" class="ui-form-preview" v-html="previewText" />
-        <span v-if="!!helpText" class="ui-form-info form-text text-muted" v-html="helpText" />
+
+        <div v-if="showPreview" class="ui-form-preview pt-1 pl-2 mt-1">{{ previewText }}</div>
+        <span v-if="Boolean(helpText)" class="ui-form-info form-text text-muted" v-html="helpText" />
```

Comment:
I was unable to find a case where setting the previewText as escaped html is different from setting the text, but would like someone to double check this, in case I unintentionally broke something,